homeassistant/components/brother/__init__.py

```diff
@@ -40,48 +34,21 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     coordinator = BrotherDataUpdateCoordinator(hass, brother)
     await coordinator.async_config_entry_first_refresh()
 
-    hass.data.setdefault(DOMAIN, {})
-    hass.data[DOMAIN].setdefault(DATA_CONFIG_ENTRY, {})
-    hass.data[DOMAIN][DATA_CONFIG_ENTRY][entry.entry_id] = coordinator
-    hass.data[DOMAIN][SNMP] = snmp_engine
+    entry.runtime_data = coordinator
+    hass.data.setdefault(DOMAIN, {SNMP: snmp_engine})
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
     return True
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_unload_entry(hass: HomeAssistant, entry: BrotherConfigEntry) -> bool:
     """Unload a config entry."""
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
 
     if unload_ok:
-        hass.data[DOMAIN][DATA_CONFIG_ENTRY].pop(entry.entry_id)
-        if not hass.data[DOMAIN][DATA_CONFIG_ENTRY]:
+        # We only want to remove the SNMP engine when unloading the last config entry
+        if len(hass.config_entries.async_entries(DOMAIN)) == 1:
```

Comment:
2 things: 1. this can be combined. 2. Doesn't async_entries return all entries? Also disabled ones?

Reply:
Oh, good point!

Reply:
Maybe double check how they did this in adguard